registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java

```diff
@@ -557,4 +559,30 @@ public int getMaxRetryCount() {
         return packet;
     }
 
+    /**
+     * Modify process name to add iteration before uploading to object store.
+     * @param regEntity
+     * @param packetKey
+     * @return
+     */
+    private String getFinalKey(SyncRegistrationEntity regEntity, String packetKey, int iteration) {
+        String[] tempKeys = packetKey.split(FORWARD_SLASH);
+        // if known format of source/process/objectName only then modify the process
+        if (tempKeys != null && tempKeys.length == 3) {
+            String source = tempKeys[0];
+            String process = tempKeys[1];
+            String objectName = tempKeys[2];
+            SubWorkflowMappingEntity workflowMappingEntity = syncRegistrationService
+                    .findWorkflowMappingByIdProcessIteration(regEntity.getRegistrationId(), process, iteration);
+
+            if (workflowMappingEntity != null &&
+                    workflowMappingEntity.getId().getAdditionalInfoReqId().equals(regEntity.getAdditionalInfoReqId())) {
+                return source + FORWARD_SLASH + process + "-" + iteration + FORWARD_SLASH + objectName;
+            } else
+                return packetKey;
+
+        } else
+            return packetKey;
```

Comment:
Since in a ideal scenario, this will not happen, can we put a warning logs here ?